### ISSUE TITLE
Enhancement offer

### ISSUE THREAD

**Issue by colepram** [GoranM/bdx#523]
Hi all,

So I got side tracked...

I was going to just use BDX as is, but I noticed this one thing that kept nagging at me. I don't really use android development, at least not yet, and constantly being nagged for an android-sdk got on my nerves. I noticed in the standard gdx-setup.jar you have options to choose sub-projects. 

I've never done any blender interface dev before and my python isn't that great, but I got it working and was wondering if you're interested in rolling up my updates.

A little explanation of what I did.

I added these check boxes to the BDX project build UI in the ui.py file
![image](https://cloud.githubusercontent.com/assets/16956852/12699817/956efa66-c7a0-11e5-9a60-2dc8fd057f97.png)

By default they're all on, so if you don't un-check any the existing behaviour is used. If you uncheck HTML, then the HTML folder isn't in the final project, uncheck android and the assets are moved to the core folder and android isn't in the final project. This seems to be consistent with what LibGDX does.

I added an "asset_folder" variable to several functions in the createproj.py file so if android isn't checked the "core/assets" folder is used instead of the "android/assets" folder

Then I heavily modified the 'replace_build_gradle' function so it updates the /build.gradle, /gradle.settings and desktop/build.gradle files if necessary to include the required sub projects

Finally I added a 'clean_project_files' function that will remove the unused desktop, android, ios, html directories, and leaves the ones that are used.

It likely needs someone to 
1) look at the code and make sure I'm not breaking any code conventions and 
2) make sure I haven't broken anything

I do get a bunch of <code>fake_module: addon missing 'bl_info' gives bad performance!:</code> errors when blender starts up. If you could tell me what that error means. Seems to only issue is I have to remember to check the enable box on the plug-in before running anything.

I've run it a number of times with different options selected and it seems to be functioning. If you're interested in this enhancement I can submit a pull request, at which point I can make changes if you have comments.

**Comment by rafcolson** [comment]
>I don't really use android development, at least not yet, and constantly being nagged for an android-sdk got on my nerves.

Seems like a lot of trouble to go into to just avoid a warning in the console. I can't say that I'm for it. Unless there might be real advantages to it.

**Comment by colepram** [comment]
Yeah, I'd considered that. I'm obsessive sometimes.

And I kind of wanted to try my hand and playing with Blender UI and Python. I've had a few work projects in the past I used Python for... and I just like wasting time solving non-issues.

But!

It does make the resulting project folder cleaner.

This:
![image](https://cloud.githubusercontent.com/assets/16956852/12699991/87a85f06-c7a7-11e5-98af-efe683c4a200.png)

Opposed to this:
![image](https://cloud.githubusercontent.com/assets/16956852/12699994/97855b40-c7a7-11e5-81bb-cf34076d3941.png)

And the big bonus, I was able to easy import the project into eclipse, which I couldn't do with standard build at all, unless I wanted to get an android SDK installed.

But I had fun doing it in any case.

**Comment by rafcolson** [comment]
I didn't know there are problems with eclipse imports in that case. If so
than I could see a real advantage.. thanks for that extra information.

**Comment by colepram** [comment]
Yeah, happens when I try to import the Gradle project

![image](https://cloud.githubusercontent.com/assets/16956852/12700077/3314b57c-c7aa-11e5-8657-f2c1d8968062.png)

Likely because it sets a default android-sdk directory if you don't specify one. Works fine under my extension. All I had to do was create the <code>Sacky.java</code> class (copied and pasted from blender code), then I could run the DesktopLauncher no problems. Same thing that shows up in blender runs under eclipse with no other changes.

It's handy to have eclipse (or some other IDE) for other logic development, but BDX is great for creating and testing models, textures, etc...

**Comment by GoranM** [comment]
The core problem here stems from the fact that `gdx-setup.jar` doesn't provide a way to set all the relevant options from the command line.

We could write a whole bunch of hacks to modify gradle files, and do whatever else is necessary to hide that fact, but that would just complicate our systems further, and to the point where it wouldn't really be justified by the few additional benefits that you outlined.

If you could contribute to libgdx, to add additional command line options to `gdx-setup.jar`, so that we can invoke it with all the settings that are currently only available in their GUI, that would really help us.

**Comment by Fr0sk** [comment]
@colepram Hey, good news. LibGDX just accepted my pull request about the command line options and it looks like in the next release (I read that it would be in a week or two) there will be an option to allow this. It's simple to use, just pass use `--excludeModules <modules>`, modules being Android, Desktop, HTML and iOS, separated by `;`. So, @colepram, can you just adapt your gui code, and clean everything else to be according with the new implementation? If there is a way, can you make the Android SDK field only appear if the Android module is selected?

**Comment by GoranM** [comment]
We also need the ability to specify which extensions to include/exclude.

It's not just about the platforms.

**Comment by Fr0sk** [comment]
> We also need the ability to specify which extensions to include/exclude.

It will produce the same project with the same extensions as now, so at least that should stay the same. But I'll add another pull request to LibGDX later today to address this issue then :)

**Comment by Fr0sk** [comment]
I made the PR for supporting extensions from the command line 8 days ago, and it hasn't got merged. Even the first I made that got merged isn't available to download yet (someone said something about the build being hanging for a day or two, but this more than a week ago. So, what do you guys think of using the unofficial build I made of gdx-setup that supports both platform and extensions from command line?

**Comment by GoranM** [comment]
I don't think this issue justifies splintering from libgdx official.

**Comment by Fr0sk** [comment]
The LibGDX Jenkins system seems to be able to build again, so the new setup is available to download. Here is the [wiki page](https://github.com/libgdx/libgdx/wiki/Project-Setup-Gradle) explaining how it works. Should be pretty straightforward. So, as long as @GoranM agrees, @colepram do you think you can change your code to reflect the new options?

**Comment by GoranM** [comment]
Yeah, I would definitely like to use the new options, so that we don't have to ship custom build.gradle files.

Thanks for getting these options into LibGDX, @Fr0sk.

**Comment by Fr0sk** [comment]
No problem, just wanted to help @GoranM. Just wondering if @colepram is still around and want to share the code though... He's been quiet for some time now.
